pkg/generator/specification.go

```diff
@@ -248,8 +248,8 @@ func (s *SpecificationGenerator) generateExecutionSpecs(function *common.Functio
 	}
 
 	runQtl, memQtl := s.determineExecutionSpecSeedQuantiles()
-	runtime := common.MinOf(common.MaxExecTimeMilli, common.MaxOf(common.MinExecTimeMilli, s.generateExecuteSpec(runQtl, runStats)))
-	memory := common.MinOf(common.MaxMemQuotaMib, common.MaxOf(common.MinMemQuotaMib, s.generateMemorySpec(memQtl, memStats)))
+	runtime := common.MinOf(common.MaxExecTimeMilli, common.MaxOf(common.MinExecTimeMilli, GenerateExecuteSpec(s.specRand, runQtl, runStats)))
```

Comment:
The function GenerateExecuteSpec seems to be used only on this package. One might want to put it private maybe.